### ISSUE TITLE
Add tags in /summon command to declaration list automatically

### ISSUE THREAD

**Issue by SPGoding** [SPGoding/datapack-language-server#641]
Suggested by NeunEinser [here](https://discordapp.com/channels/666020457568403505/666037123450929163/742889249619181640) in the Discord server.

Tags in `/summon` command's `Tags` tag should be treated as public declarations, unless that tag has already been declared with a different visibility in the scope.

**Comment by NeunEinser** [comment]
If at all possible, tags created like that could default to private if they are removed in the same file, or within if other functions are called between creation and removal.

If that's too complicated it should be good to overwrite the visibility manually using declare in those cases, it just would be a nice detail 😉

**Comment by NeunEinser** [comment]
By now, I have a different opinion on this.
While it might be generally a good default to register tags used in /summon or /tag ... add, I think it should be configurable. 

By now, I actually came to the conclusion that documenting your tags using #declare might not be the worst of ideas. 
The obvious difference from e.g. /scoreboard objectives add is that for tags, there is no command that you execute only once to declare a tag. You may use different /summon or /tag add commands including the same tag in different places.
/scoreboard objectives add is only done once per objective, so that command can cleanly serve as an implicit declaration. 

I think it would be a shame, if in a datapack which uses #declare for tags, sanity checking tags in these places was lost.

It might be worthwhile to add a similar setting for score holders, where they do auto complete in all places and are sanity checked wherever they're only being read. But also as an option, to not lose sanity checking in case the packer already uses #declare.

**Comment by SPGoding** [comment]
Yes, what I have in mind is having some options like `implicitXDeclaration` (similar to the `strictXCheck` family)
